queasars/minimum_eigensolvers/base/termination_criteria.py

```diff
@@ -80,36 +94,51 @@ class BestIndividualRelativeChangeTolerance(EvolvingAnsatzMinimumEigensolverBase
     :param minimum_relative_change: relative improvement in expectation value below which the algorithm shall
         terminate. Must be in the range )0,1)
     :type minimum_relative_change: float
+    :param allowed_consecutive_violations: determines how often the threshold value can be violated consecutively
+        before this criterion chooses to terminate. If set to 0, this terminates the first time the change falls
+        below the threshold value. If set to 2 for example, this terminates the first time the change falls below
+        the threshold three consecutive times. Must be at least 0.
+    :type allowed_consecutive_violations: int
     """
 
-    def __init__(self, minimum_relative_change: float):
+    def __init__(self, minimum_relative_change: float, allowed_consecutive_violations: int = 0):
         if minimum_relative_change <= 0 or minimum_relative_change > 1:
             raise ValueError("The minimum relative improvement parameter must not exceed the range )0,1)!")
+        if allowed_consecutive_violations < 0:
+            raise ValueError("allowed_consecutive_violations must be at least 0!")
 
-        self._minimum_relative_change = minimum_relative_change
-        self._previous_expectation_value: float = float("inf")
+        self._minimum_relative_change: float = minimum_relative_change
+        self._allowed_consecutive_violations: int = allowed_consecutive_violations
+        self._previous_expectation_value: Optional[float] = None
+        self._relative_change_history: list[float] = []
 
     def reset_state(self) -> None:
-        self._previous_expectation_value = float("inf")
+        self._previous_expectation_value = None
+        self._relative_change_history = []
 
     def check_termination(
         self,
         population_evaluation: BasePopulationEvaluationResult,
         best_individual: BaseIndividual,
         best_expectation_value: float,
     ) -> bool:
-        if self._previous_expectation_value == float("inf"):
+
+        if self._previous_expectation_value is None:
             self._previous_expectation_value = population_evaluation.best_expectation_value
             return False
 
         relative_change = abs(self._previous_expectation_value - population_evaluation.best_expectation_value) / abs(
             self._previous_expectation_value
         )
-        if relative_change >= self._minimum_relative_change:
-            self._previous_expectation_value = population_evaluation.best_expectation_value
+        self._previous_expectation_value = population_evaluation.best_expectation_value
+        self._relative_change_history.append(relative_change)
+
+        if len(self._relative_change_history) < self._allowed_consecutive_violations + 1:
             return False
 
-        return True
+        max_change = max(self._relative_change_history[-self._allowed_consecutive_violations - 1 :])
+
+        return not max_change >= self._minimum_relative_change
```

Comment:
Similar as above, I would suggest to invert the inequality.

Reply:
Done :).